src/modules/httptrap.c

```diff
@@ -1201,14 +1219,20 @@ rest_httptrap_handler(mtev_http_rest_closure_t *restc,
   }
   mtev_http_response_standard(ctx, error_code, "ERROR", "application/json");
   mtev_http_response_append(ctx, "{ \"error\": \"", 12);
-  if (rxc && rxc->error)
+  if (rxc && rxc->error) {
     error = rxc->error;
+  }
   mtevL(nldeb, "Error %s for %s (%" PRIu64 ")\n", error, npats ? pats[0] : "?", current_counter);
   yajl_string_encode((yajl_print_t)http_write_encoded, ctx, (const unsigned char*)error, strlen(error), 0);
   mtev_http_response_append(ctx, "\" }", 3);
   mtev_http_response_end(ctx);
   mtev_memory_end();
   mtevAssert(!mtev_memory_in_cs());
+  if (rxc) {
+    if (rxc->immediate_metrics && mtev_hash_size(rxc->immediate_metrics)) {
```

Comment:
You have `mtev_hash_size(json->immediate_metrics) != 0` above.  This should match that.